sdk/python/feast/permissions/auth/token_parser.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+
+class TokenParser(ABC):
+    """
+    A class to parse an access token to extract the user credential and roles.
+    """
+
+    @abstractmethod
+    async def user_details_from_access_token(
+        self, access_token: str, **kwargs
+    ) -> tuple[str, list[str]]:
+        """
+        Parse the access token and return the current user and the list of associated roles.
+
+        Returns:
```

Comment:
so, instead of a `RoleManager` we need a `UserManager` with `get_users` and add `has_roles` method to `User` itself?
it would require some reviews of the existing code, but it's doable

Reply:
I mean return User instance instead of `tuple[str, list[str]]` 

 ```
async def user_details_from_access_token(
        self, access_token: str, **kwargs
    ) -> User:
```

and in implementation we can set  and return `User(username=current_user, roles=roles)`

Reply:
Changed to introduce a `User` class and totally removed the `RoleManager`, as it's not needed until we don't want to cache the users.
A bit simpler now, but many changes were committed.